staging_service/app.py

```diff
@@ -36,6 +95,70 @@ async def importer_mappings(request: web.Request) -> web.json_response:
     return web.json_response(data=mappings)
 
 
+def _file_type_resolver(path: PathPy) -> FileTypeResolution:
+    fi = AutoDetectUtils.get_mappings([str(path)])["fileinfo"][0]
+    # Here we assume that the first entry in the file_ext_type field is the entry
+    # we want. Presumably secondary entries are less general.
+    ftype = fi['file_ext_type'][0] if fi['file_ext_type'] else None
+    if ftype in _IMPSPEC_FILE_TO_PARSER:
+        return FileTypeResolution(parser=_IMPSPEC_FILE_TO_PARSER[ftype])
+    else:
+        ext = fi['suffix'] if fi['suffix'] else path.suffix[1:] if path.suffix else path.name
+        return FileTypeResolution(unsupported_type=ext)
+
+
+def _format_errors(errors: tuple[Error, ...], paths: dict[PathPy, Path]) -> str:
+    errs = []
+    for e in errors:
+        file1 = None
+        tab1 = None
+        file2 = None
+        tab2 = None
+        if e.source_1:
+            file1 = paths[e.source_1.file].user_path
+            tab1 = e.source_1.tab
+        if e.source_2:
+            file2 = paths[e.source_2.file].user_path
+            tab2 = e.source_2.tab
+        errs.append(_ERROR_FORMATTERS[e.error](e.message, file1, tab1, file2, tab2))
+    return json.dumps({"errors": errs})
+
+
+@routes.get("/bulk_specification/{query:.*}")
+async def bulk_specification(request: web.Request) -> web.json_response:
+    """
+    Takes a `files` query parameter with a list of comma separated import specification file paths.
+    Returns the contents of those files parsed into a list of dictionaries, mapped from the data
+    type, in the `types` key.
+    
+    :param request: contains a comma separated list of files, e.g. folder1/file1.txt,file2.txt
+    """
+    username = await authorize_request(request)
+    files = parse_qs(request.query_string).get("files", [])
+    files = files[0].split(",") if files else []
+    files = [f.strip() for f in files if f.strip()]
+    paths = {}
+    for f in files:
+        p = Path.validate_path(username, f)
+        paths[PathPy(p.full_path)] = p
+    # list(dict) returns a list of the dict keys in insertion order (py3.7+)
+    res = parse_import_specifications(
+        tuple(list(paths)),
+        _file_type_resolver,
+        lambda e: logging.error("Unexpected error while parsing import specs", exc_info=e))
```

Comment:
I tested the OTHER error pathway and this log print by temporarily introducing an error into the parser, but I don't see a reasonable way to test it in an integration test like this. The parser is tested with unit tests though.